### ISSUE TITLE
[Bug] [CameraView] [Android] Flash does not work

### ISSUE THREAD

**Issue by acuntex** [xamarin/XamarinCommunityToolkit#301]
### Description
On Android the CameraView does not trigger a flashlight even though Flash is on.

### Steps to Reproduce

1. Use the sample app.
2. Let Flash on.
3. Trigger shutter.

### Expected Behavior
The flash light works.

### Actual Behavior
The flash light does not works.

### Reproduction imagery
As you can see, the can does not reflect any flash light:
![Screenshot_20200913_175908_com companyname xamarincommunitytoolkitsample](https://user-images.githubusercontent.com/17462039/93022836-7eb2fb00-f5eb-11ea-94a7-cf1fe7a3e274.jpg)

**Comment by LucioMSP** [comment]
I will be taking responsibility for this issue. #Hacktoberfest

**Comment by jfversluis** [comment]
Great @LucioMSP ! Any idea on when we can expect something? :) Let us know how it goes!

**Comment by jfversluis** [comment]
@LucioMSP unless I hear otherwise from you that you are actively working on it, I am going to allow other to take this issue

**Comment by bmacombe** [comment]
@jfversluis I'll investigate this one.

**Comment by bmacombe** [comment]
For me at least, the flash works on the second capture after it's turned on.  Looking into why at first glance the code looks correct.

**Comment by acuntex** [comment]
@bmacombe On a Huawei P30 Pro the flash never works.

**Comment by bmacombe** [comment]
Yeah it seems the behavior varies by device.  I was using moto g5.  @pictos tried on another device and it would flash on the first capture and then not after.

**Comment by pictos** [comment]
FYI I'm using Motorola one with Android 10

**Comment by m3mo89** [comment]
@pictos @bmacombe @jfversluis in the XamarinCommunityToolkitSample when you activate the flash the following event in **CameraViewPage** assigns CameraFlashMode.**On** as the flash mode

```
void FlashSwitch_Toggled(object sender, ToggledEventArgs e)
    => cameraView.FlashMode = e.Value ? CameraFlashMode.On : CameraFlashMode.Off;
```
then according to the **SetFlash** method in the **CameraFragment** class, the real flash mode assigned would be **Single**

```
flashMode = Element.FlashMode switch
{
    CameraFlashMode.Off => FlashMode.Off,
    CameraFlashMode.Torch => FlashMode.Torch,
    _ => FlashMode.Single,
};
```
So I want to think that having Single mode as Flash mode which only fires the flash once instead of Torch mode which does continuous flash is what is causing this bug.

**Comment by bmacombe** [comment]
I just got a Moto G7 with Android 9 going, I get the same as Moto One with Android 10.  Flash works on the first capture, stops working after that.

**Comment by pictos** [comment]
@m3mo89, tks for investigate this issue. Our guess is that the problem is after the first set-up. So we need to investigate a little more, this `flashMode` is correct based on the docs, it's just a enum. Also, from the docs we have two ways to configure the camera flash so we are investigating about that. If you have any news please let us know.

**Comment by jfversluis** [comment]
What is the status on this one? Can anyone tell me? :)

**Comment by bmacombe** [comment]
The status is, it's a mess :) Tried a lot of different strategies found online, but still would get varying results on different versions of Android, devices, etc.

**Comment by DevarajSekar** [comment]
Hi Team / @jfversluis ,

Greetings, Any update/help on this issue much appreciated.

**Comment by DevarajSekar** [comment]
Hi @jfversluis ,

Any help on the request.

Regards,
Devaraj S
